test/compatibility/test_wt10533.py

```diff
@@ -0,0 +1,152 @@
+#!/usr/bin/env python
+#
+# Public Domain 2014-present MongoDB, Inc.
+# Public Domain 2008-2014 WiredTiger, Inc.
+#
+# This is free and unencumbered software released into the public domain.
+#
+# Anyone is free to copy, modify, publish, use, compile, sell, or
+# distribute this software, either in source code form or as a compiled
+# binary, for any purpose, commercial or non-commercial, and by any
+# means.
+#
+# In jurisdictions that recognize copyright laws, the author or authors
+# of this software dedicate any and all copyright interest in the
+# software to the public domain. We make this dedication for the benefit
+# of the public at large and to the detriment of our heirs and
+# successors. We intend this dedication to be an overt act of
+# relinquishment in perpetuity of all present and future rights to this
+# software under copyright law.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+# EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+# MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.
+# IN NO EVENT SHALL THE AUTHORS BE LIABLE FOR ANY CLAIM, DAMAGES OR
+# OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE,
+# ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+# OTHER DEALINGS IN THE SOFTWARE.
+
+import compatibility_test, os, shutil, sys
+
+class test_wt10533(compatibility_test.CompatibilityTestCase):
+    '''
+    Test handling checkpoints during database downgrade.
+    '''
+
+    conn_config = 'compatibility=(release="3.3"),' + \
+        'statistics=(fast),statistics_log=(json,on_close,wait=5)'
+    create_config = 'key_format=i,value_format=S'
+    uri = 'table:test_wt10533'
+
+    # Initial pass
+    num_initial_rows = 100
+    num_checkpoints = 100
+
+    # Second pass
+    num_second_rows_per_checkpoint = 10000
+    num_second_checkpoints = 10
+
+    def test_checkpoint_downgrade(self):
+        '''
+        Test downgrading after creating a large number of checkpoints, which forces the
+        monotonic checkpoint time to race far ahead the wallclock time.
+        '''
+
+        # Checkout the relevant branches
+        branch = 'mongodb-6.0'
+        self.prepare_wt_branch(branch)
+        self.prepare_wt_branch('develop')
+
+        # Now run the first part of the test on the develop branch.
+        self.run_method_on_branch('develop', 'on_develop_test_checkpoint_downgrade')
+
+        # Now run the second part of the test on the other branch.
+        self.run_method_on_branch(branch, 'on_branch_test_checkpoint_downgrade')
+
+    def on_develop_test_checkpoint_downgrade(self):
+        '''
+        The second part of the test, which runs on a different branch.
```

Comment:
Isn't this the first part of the test?

Reply:
Oh yes, too much of mindless copy and paste...